helpers.php

```diff
@@ -1065,6 +1255,85 @@ function refundOrReleaseTransactionByStatus($transaction)
     $api->call();
 }
 
+/**
+ * Get the terminal ID based on the remote name and shop ID.
+ *
+ * @param string $remote_name
+ * @param int $shop_id
+ *
+ * @return int|string
+ */
+function getTerminalIdByRemoteName($remote_name, $shop_id = 1)
+{
+    $terminal_id = '';
+    try {
+        if (filter_var($shop_id, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1]]) === false) {
+            throw new Exception('Invalid shop id');
+        }
+        $query = 'SELECT id_terminal FROM `' . _DB_PREFIX_ . "altapay_terminals` WHERE active = 1 AND remote_name = '" . pSQL($remote_name) . "' AND shop_id = '" . $shop_id . "' ";
```

Comment:
Same comments as above.

Reply:
Integer validation has already been applied to Line1270 above.